File: Leetcode (Python, JavaScript)/876. Middle of the Linked List/876. Middle of the Linked List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    let len = 0;
    let curr = head;
    while (curr!==null){
        len += 1;
        curr = curr.next;
    }
    let curr1 = head;
    for (let i=0;i<Math.floor(len/2);i++){
        curr1 = curr1.next;
    }
    return curr1;
};

File: Leetcode (Python, JavaScript)/876. Middle of the Linked List/876. Middle of the Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        len = 0
        curr = head
        while curr is not None:
            len += 1
            curr = curr.next
        curr = head
        for i in range(len//2):
            curr = curr.next
        return curr